### PR TITLE
sql: fix ARRAY serialization

### PR DESCRIPTION
**Pull request by RaduBerinde**
We were formatting ARRAY as `{1,2}` which is not valid when we are serializing.
Switching to `ARRAY[1,2]` in this case.

@arjunravinarayan this fixes the `TestPGPreparedQuery` failure when distsql is in auto mode.

### CODE REVIEW

**Comment by arjunravinarayan** [comment]
:lgtm:

Nice quick triage, thanks!

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15462#-:-KiowpD_N8w381DHS1BU:b-eb8vw5)*

File: pkg/sql/parser/datum.go [hunk 1]
```diff
@@ -2001,14 +2001,22 @@ func (*DArray) AmbiguousFormat() bool { return false }
 
 // Format implements the NodeFormatter interface.
 func (d *DArray) Format(buf *bytes.Buffer, f FmtFlags) {
-	buf.WriteByte('{')
+	if f.disambiguateDatumTypes {
+		buf.WriteString("ARRAY[")
```
Comment (knz) [inline pkg/sql/parser/datum.go]: As I said on the other issue I would just go for the `ARRAY[...]` notation all the time. The code would become simpler.

**Comment by knz** [comment]
(LGTM either way - you decide)

**Comment by RaduBerinde** [comment]
Done, PTAL

**Comment by knz** [comment]
:lgtm:

---

Reviewed 8 of 8 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15462#-:-Kip_T8GgwNuO86PwXQU:bnfp4nl)*

**Pull request merged**